provider/clusterrosahcp/upgrade/cluster_upgrade.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package upgrade
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	semver "github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/terraform-redhat/terraform-provider-rhcs/provider/common"
+	"github.com/zgalor/weberr"
+)
+
+// ClusterUpgrade bundles the description of the upgrade with its current state
+type ClusterUpgrade struct {
+	policy      *cmv1.UpgradePolicy
+	policyState *cmv1.UpgradePolicyState
+}
+
+func (cu *ClusterUpgrade) State() cmv1.UpgradePolicyStateValue {
+	return cu.policyState.Value()
+}
+
+func (cu *ClusterUpgrade) Version() string {
+	return cu.policy.Version()
+}
+
+func (cu *ClusterUpgrade) NextRun() time.Time {
+	return cu.policy.NextRun()
+}
+
+func (cu *ClusterUpgrade) Delete(ctx context.Context, client *cmv1.ClustersClient) error {
+	_, err := client.Cluster(cu.policy.ClusterID()).UpgradePolicies().UpgradePolicy(cu.policy.ID()).Delete().SendContext(ctx)
```

Comment:
should be something like https://github.com/openshift/rosa/blob/master/pkg/ocm/controlplane_upgrades.go#L21-L29